### ISSUE TITLE
Hour picker - 01 am, 01 pm ain't rendered properly on android

### ISSUE THREAD

**Issue by KyoungHwanLee** [henninghall/react-native-date-picker#382]
**Describe the bug**
On android + 12hour mode, when set hour value as 01 am or 01pm, the hour item rendered as 12.
<img width="329" alt="s0" src="https://user-images.githubusercontent.com/7302920/132156683-ce2b012a-aefb-405f-b568-3d36e51a5e57.png">

When picker view is re-rendered (ex. scroll) , it is updated to proper value.
<img width="325" alt="s1" src="https://user-images.githubusercontent.com/7302920/132157260-211a3cc8-310f-4695-966e-6965fed2bdae.png">

It happens when `androidVariant = nativeAndroid`.

**Expected behavior**
Hour label text should be 1

**To Reproduce**
```
const date = new Date();
date.setHours(13);
// or date.setHours(1);

return (
    <DatePicker
        modal
        date={date}
        open={true}
        mode="time"
        locale="ko"
    />
)
```

**Smartphone (please complete the following information):**
 - OS: Android
 - react-native-date-picker version 4.0.0

**Comment by henninghall** [comment]
I wonder if it might have to do with you mutating the date, what happens if you write 
```
const date = new Date("1300")
```
instead?

**Comment by KyoungHwanLee** [comment]
@henninghall Sorry for my late update.
`const date = new Date("1300")` Just set year to 1300.
I think it's not has to do with javascript side, maybe something has gone wrong on Android native view.

**Comment by KyoungHwanLee** [comment]
And I found a weird behavior and a dirty solution.

https://github.com/henninghall/react-native-date-picker/blob/79e1e7501e2910300827433eca5cebecb02849e2/android/src/main/java/com/henninghall/date_picker/wheels/Wheel.java#L94-L100

When I modify 97th line to
`displayValues.add("_" + this.toDisplayValue(value));`

<img width="360" alt="pre" src="https://user-images.githubusercontent.com/7302920/132624386-4b340221-c33d-4b92-976e-374f6868bceb.png">

Nothing changed.
But when use
`displayValues.add(this.toDisplayValue(value) + "_");`
<img width="329" alt="post" src="https://user-images.githubusercontent.com/7302920/132624464-f4b0d725-a167-4ac0-b66e-a252c62b4607.png">
1am/pm rendered properly.

So, finally
`displayValues.add(" " + this.toDisplayValue(value) + " ");`
<img width="348" alt="both" src="https://user-images.githubusercontent.com/7302920/132624528-cc48735d-6f65-4574-b383-fd319d3e385b.png">
fixed my issue, but definitely not a good solution.

Is there any code like `labelText.endsWith(numberValue)`?

**Comment by hkbok** [comment]
I am running into the same problem. Will it take a long time to resolve?

**Comment by peterphan1996** [comment]
I also have the same issue

**Comment by hwanginyeop** [comment]
When date props is `moment('2021-11-12 13:37:00').toDate()`,
in picker, date display `PM(오후) 12:37`.

Also when date props is `moment('2021-11-12 01:37:00').toDate()`,
in picker, date display `AM(오전) 12:37`.

**Comment by PreetamJakhmola1** [comment]
Facing the same issue. but thanks to @KyoungHwanLee  for giving this temporary solution. love you man.

**Comment by henninghall** [comment]
A fix for this is released in v4.1.5
